File: Difficulty: Medium/Stickler Thief/stickler-thief.cpp
class Solution {
  public:
    int findMaxSum(vector<int>& arr) {
        // code here
        int pos1 = arr[0];
        int pos2 = 0;
        int i = 1;
        int n = arr.size();
        int ans = 0;
        while(i<n) {
            ans = max(pos2+arr[i],pos1);
            pos2 = pos1;
            pos1 = ans;
            i++;   
        }
        
        return ans;
    }
};

File: Difficulty: Medium/Stickler Thief/stickler-thief.java
class Solution {
    
    private int solve(int[] arr, int index, int n,int[] visited) {
        if(index>=n) {
            return 0;
        }
        
        if(visited[index]!=0) return visited[index];
        
        int missing = solve(arr,index+1,n,visited);
        int including = arr[index]+solve(arr,index+2,n,visited);
        
        visited[index] =  Math.max(missing,including);
        return visited[index];
        
    }
    
    public int findMaxSum(int arr[]) {
        // code here
        int n = arr.length;
        
        int index = 0;
        int sum1 = 0;
        int sum2 = 0;
        
        int[] visited = new int[n];
        
        int ans = solve(arr, index, n,visited);
        return ans;
        //return Math.max(sum1,sum2);
        
        
    }
}